src/org/linphone/IncomingCallActivity.java

```diff
@@ -207,16 +207,27 @@ protected void onResume() {
 		LinphoneAddress address = mCall.getRemoteAddress();
 		// May be greatly sped up using a drawable cache
 		Contact contact = ContactsManager.getInstance().findContactWithAddress(getContentResolver(), address);
-		LinphoneUtils.setImagePictureFromUri(this, mPictureView.getView(), contact != null ? contact.getPhotoUri() : null,
+		LinphoneUtils.setImagePictureFromUri(this, mPictureView, contact != null ? contact.getPhotoUri() : null,
 				contact != null ? contact.getThumbnailUri() : null, R.drawable.unknown_small);
 
-		// To be done after findUriPictureOfContactAndSetDisplayName called
-		mNameView.setText(contact != null ? contact.getName() : "");
-		if (getResources().getBoolean(R.bool.only_display_username_if_unknown)) {
-			mNumberView.setText(address.getUserName());
+
+		if (contact == null) {
+			if (getResources().getBoolean(R.bool.only_display_username_if_unknown) && LinphoneUtils.isSipAddress(address.asStringUriOnly())) {
+				mNameView.setText(address.getUserName());
+				//partnerName.setText(contactName);
+			} else {
+				mNameView.setText(address.asStringUriOnly());
+				// partnerName.setText(contactName);
+			}
 		} else {
-			mNumberView.setText(address.asStringUriOnly());
+			mNameView.setText(contact.getName());
 		}
+		// To be done after findUriPictureOfContactAndSetDisplayName called
+		//TODO: question????lame_if_unknown)) {
+//			mNumberView.setText(address.getUserName());
+//		} else {
+//			mNumberView.setText(address.asStringUriOnly());
+//		}
```

Comment:
It looks like the above commented-out lines are actually included above. Can we cut these out?